Lib/asyncio/selector_events.py

```diff
@@ -362,13 +364,14 @@ async def sock_recv(self, sock, n):
             pass
         fut = self.create_future()
         fd = sock.fileno()
-        self.add_reader(fd, self._sock_recv, fut, sock, n)
+        handle = self.add_reader(fd, self._sock_recv, fut, sock, n)
```

Comment:
Adding to the comment above, maybe you should use `self._add_reader` here and fix `loop.add_reader` to return `None` (and not the result of `_add_reader`)

Reply:
Right, that works too. API change is a bit concerning for sure.